communication.md

```diff
@@ -10,6 +10,24 @@ The SDK **MUST** use the [Telemetry ingest APIs](https://docs.newrelic.com/docs/
 * Only send API keys as headers (not query params)
 
 
+### Request ID header
+
+When communicating with data ingest services, there are 3 possible outcomes of the HTTP call:
+
+1. OK status (200 <= status_code < 300), indicating data has been received and persisted.
+2. non-OK status, indicating data has not been persisted
+3. disconnect (either client or server). In this case, the connection is closed prior to receiving a status indication.
+
+In case (3) above, the client should only retry the request if the request is idempotent since data may or may not have been persisted (and thus the data may get recorded twice, resulting in the data aggregates being inaccurate).
+
+To prevent data loss while allowing clients to retransmit in the case of transient failures, the ingest service must be able to identify duplicate requests; therefore, all SDKs **MUST** send the following HTTP header with the request:
+
+| Header Name | Header Value | Code Example |
+| ----------- | ------------ | ------------ |
+| x-request-id | A [version 4 UUID string](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) | [str(uuid.uuid4())](https://docs.python.org/3/library/uuid.html#uuid.uuid4) |
+
+**NOTE** the request ID should be generated at the start of a request "cycle" and the value should be maintained throughout any retries.
```

Comment:
We haven't used the term _request cycle_ before.
I take it to mean that for a given payload, the same request ID should be used for all subsequent retries.  

This is complicated by splitting retries.  If the server responds with a 413, and the SDK splits the payload to retry, those retried requests should actually each get a new request ID.

Reply:
Ah great point! Splitting payloads would generate a new request ID. I'll make that a little bit clearer in the wording of the spec!